client/js/lounge.js

```diff
@@ -224,14 +224,40 @@ $(function() {
 			}
 		};
 
-		$(window).on("focus", focus);
+		$(document.body).on("keydown", (e) => {
+			// If user is typing into some field, don't process
+			if (e.target !== document.body) {
+				return;
+			}
 
-		chat.on("click", ".chat", function() {
-			setTimeout(function() {
-				if (window.getSelection().isCollapsed) {
-					focus();
-				}
-			}, 2);
+			// Do not focus when using alt or meta (windows) keys
+			if (e.altKey || e.metaKey) {
+				return;
+			}
+
+			// Ignore shift key press if it's not combined
+			if (e.shiftKey && e.which === 16) {
+				return;
+			}
+
+			// Ignore all ctrl keys except for ctrl+v to allow pasting
+			if (e.ctrlKey && e.which !== 86) {
```

Comment:
This doesn't work on macOS 😭

My understanding is that `metaKey` is <kbd>cmd/⌘</kbd> on Mac (on Mac copy/pasting is ⌘C ⌘V, not Ctrl+C Ctrl+V, in fact, _most_ things are ⌘ instead of Ctrl).

Alternatively, can you make use of Mousetrap for this code? It seems to handle the Ctrl/Cmd/Alt mess pretty well.

Reply:
Damn macs! I can't really use mousetrap here because the code is very specific and uses `keydown` event.

I'll see what I can do.

Reply:
Should be fixed now.

Reply:
Yep, works perfectly.